handlers/api/commentApiHandler.js

```diff
@@ -19,6 +19,24 @@ CommentApiHandler.addComment = function (req, res) {
         });
 };
 
+CommentApiHandler.updateComment = function (req, res) {
+    comments.update(
+        sanitizer(req.body.commentId),
+        sanitizer(req.body.comment),
+        sanitizer(req.body.software_version_id),
+
+        function (result, error) {
+            comments.getById(sanitizer(req.body.commentId), function(comment) {
+                // if not the owner of this comment and not an admin
+                if (comment.userid != req.user.id && req.user.role != 0) {
```

Comment:
Generally its safer to use "!==" instead of "!=" (and '===' instead of '=='). In this case if a user by some chance had a unset role and if the DB handler interpreted that as empty string '' the user might get interpreted as an admin.